styles/main.css

```diff
@@ -122,30 +123,37 @@ nav ul li{
 }
 
 nav {
-    padding: 8px 2em;
-    border: 0 0 1px 0 dotted rgb(0,0,0,0.3);
-    margin: -4px 0 0 0 
+    border-bottom: 1px dotted rgb(0,0,0,0.3);
+    margin: -4px 2em 0 2em;
+    overflow: auto;
 }
 
 header {
-    border: 4px 0 0 0 solid rgb(0,0,0,0.1);
+    border-top: 4px solid rgb(0,0,0,0.1);
     margin: 0 -2em;    
 }
 
-nav ul a:hover {
+nav ul a {
     padding:1em 0 0.5em 0;
     margin: 0 1em;
-    display:block;
-    border: 4px 0 0 0 solid;
-    border:4px solid rgba(255,255,255,0);
+    display:block;    
+    border-top: 4px solid rgba(255,255,255,0);
+}
+nav ul a:hover {
+    border-top: 4px solid rgba(255,255,255,1);
 }

```

Comment:
You need two separate rules here:  one for the hyperlinks in their default state, where you set padding, margin, display, etc; then a second to specify any changes when the hyperlinks are hovered (in this case only border-color)